src/openrct2/world/MapGen.cpp

```diff
@@ -167,52 +172,44 @@ void mapgen_generate(mapgen_settings *settings)
 
     // Initialise the base map
     map_init(mapSize);
-    for (y = 1; y < mapSize - 1; y++) {
-        for (x = 1; x < mapSize - 1; x++) {
+    for (y = 1; y < mapSize - 1; y++)
+    {
+        for (x = 1; x < mapSize - 1; x++)
+        {
             tileElement = map_get_surface_element_at(x, y);
             tile_element_set_terrain(tileElement, floorTexture);
             tile_element_set_terrain_edge(tileElement, wallTexture);
-            tileElement->base_height = settings->height;
+            tileElement->base_height      = settings->height;
             tileElement->clearance_height = settings->height;
         }
     }
 
     // Create the temporary height map and initialise
     _heightSize = mapSize * 2;
-    _height = (uint8*)malloc(_heightSize * _heightSize * sizeof(uint8));
+    _height     = (uint8 *) malloc(_heightSize * _heightSize * sizeof(uint8));
```

Comment:
Not sure how thorough you want to be with this PR - but ideally this would now be `new uint8[_heightSize * _heightSize]`

Reply:
While we're changing it, why not just put them on stack and let compiler figure out if it can optimise it further?

Reply:
What do you mean?

Reply:
This one is not the best example, but you suggested the `grassTreeIds` be made `std::vector`. I don't even see the point of them being a vector, just make them an array `sint32 grassTreeIds[Util::CountOf(GrassTrees)];`.

`_height` could also use the same treatment, but it would have to be static variable in the scope of the file – or better yet, be passed into functions that use it via pointer.

Reply:
The vector is better because it means you don't then need a separate count variable. The count of grass trees is more of a max capacity rather than fixed count.

I can't remember how height is used so I am not sure how I would go about doing that if I was to write it now.

Reply:
Could you two decide on what to do here? I don't know enough about C++ myself to say what's best.

Reply:
@IntelOrca my point was to reduce allocations required and potentially having compiler figure things out when there is no aliasing possible. With vector you go the other way, potentially even doing memory copies if it has to reallocate, and change the code using it just to save the count variable.

I would have gone `sint32 grassTreeIds[Util::CountOf(GrassTrees)];`, as it is simpler, doesn't require any other changes (but for removing `free`), _may_ help the compiler. The drawback is you are putting large things on stack.

@IntelOrca the decision is up to you.

Reply:
Prefer `std` and only avoid it if we actually hit performance issues.

Since this is a one off things as part of a map generation, it doesn't need to be micro optimised.